src/auth0-session/client/edge-client.ts

```diff
@@ -0,0 +1,241 @@
+    let result: ReturnType<typeof oauth.validateAuthResponse>;
+    try {
+      result = oauth.validateAuthResponse(as, client, url, expectedState);
+    } catch (e) {
+      throw new ApplicationError(e);
+    }
+    if (oauth.isOAuth2Error(result)) {
+      throw new IdentityProviderError({
+        message: result.error_description || result.error,
+        error: result.error,
+        error_description: result.error_description
+      });
+    }
+    return result;
+  }
+
+  async callback(
+    redirectUri: string,
+    parameters: URLSearchParams,
+    checks: OpenIDCallbackChecks,
+    extras: CallbackExtras
+  ): Promise<TokenEndpointResponse> {
+    const [as, client] = await this.getClient();
+
+    const { clientAssertionSigningKey, clientAssertionSigningAlg } = this.config;
+
+    let clientPrivateKey = clientAssertionSigningKey as CryptoKey | undefined;
+    /* c8 ignore next 3 */
+    if (clientPrivateKey && !(clientPrivateKey instanceof CryptoKey)) {
+      clientPrivateKey = await jose.importPKCS8<CryptoKey>(clientPrivateKey, clientAssertionSigningAlg || 'RS256');
+    }
+    const response = await oauth.authorizationCodeGrantRequest(
+      as,
+      client,
+      parameters,
+      redirectUri,
+      checks.code_verifier as string,
+      {
+        additionalParameters: extras.exchangeBody,
+        ...(clientPrivateKey && { clientPrivateKey }),
+        ...this.httpOptions()
+      }
+    );
+
+    const result = await oauth.processAuthorizationCodeOpenIDResponse(
+      as,
+      client,
+      response,
+      checks.nonce,
+      checks.max_age
+    );
+    if (oauth.isOAuth2Error(result)) {
+      throw new IdentityProviderError({
+        message: result.error_description || /* c8 ignore next  */ result.error,
+        error: result.error,
+        error_description: result.error_description
+      });
+    }
+    return result;
+  }
+
+  async endSessionUrl(parameters: EndSessionParameters): Promise<string> {
+    const [as] = await this.getClient();
+    const issuerUrl = new URL(as.issuer);
+
+    if (
+      this.config.idpLogout &&
+      (this.config.auth0Logout || (issuerUrl.hostname.match('\\.auth0\\.com$') && this.config.auth0Logout !== false))
+    ) {
+      const { id_token_hint, post_logout_redirect_uri, ...extraParams } = parameters;
+      const auth0LogoutUrl: URL = new URL(urlJoin(as.issuer, '/v2/logout'));
+      post_logout_redirect_uri && auth0LogoutUrl.searchParams.set('returnTo', post_logout_redirect_uri);
+      auth0LogoutUrl.searchParams.set('client_id', this.config.clientID);
+      Object.entries(extraParams).forEach(([key, value]: [string, string]) => {
+        if (value === null || value === undefined) {
+          return;
+        }
+        auth0LogoutUrl.searchParams.set(key, value);
+      });
+      return auth0LogoutUrl.toString();
+    }
+    if (!as.end_session_endpoint) {
+      throw new Error('no rp inititated logout');
```

Comment:
Should this error have a more specific type?

Reply:
None of our configuration errors have a type, they just need to tell the developer they've configured the SDK wrong

Reply:
Should this error in particular be a little more descriptive then? If I were a developer not versed in identity and got that error, I think I'd be confused.

Reply:
Sure - I can improve the message